src/storage/test/StorageLookupBenchmark.cpp

```diff
@@ -0,0 +1,304 @@
+    std::shared_ptr<nebula::cpp2::IndexItem> index;
+    BENCHMARK_SUSPEND {
+        std::string rootPath;
+        rootPath = folly::stringPrintf("%s/%s_%d",
+                                       FLAGS_root_data_path.c_str(),
+                                       "Precise", expectRowNum);
+        kv = TestUtils::initKV(std::move(rootPath).c_str());
+        schemaMan = mockSchemaMan();
+        indexMan = mockIndexMan();
+        auto iRet = indexMan->getTagIndex(spaceId, indexId);
+        if (!iRet.ok()) {
+            LOG(ERROR) << "Index get error";
+            return;
+        }
+        index = iRet.value();
+        if (!genData(kv.get(), schemaMan.get(), index)) {
+            LOG(ERROR) << "Data load error";
+            return;
+        }
+    };
+    auto ret = processLookup(kv.get(), schemaMan.get(), indexMan.get(), filter, expectRowNum);
+    if (!ret) {
+        LOG(ERROR) << "Processor error";
+        return;
+    }
+}
+
+void lookupPrecise(int32_t expectRowNum) {
+    auto* col = new std::string("col_0");
+    auto* alias = new std::string("col0");
+    auto* ape = new AliasPropertyExpression(new std::string(""), alias, col);
+    std::string c0("col0_row_1");
+    auto* pe = new PrimaryExpression(c0);
+    auto r =  std::make_unique<RelationalExpression>(ape,
+                                                     RelationalExpression::Operator::EQ,
+                                                     pe);
+    lookupExec(expectRowNum, Expression::encode(r.get()));
+}
+
+void lookupFilter(int32_t expectRowNum) {
+    auto* col = new std::string("col_1");
+    auto* alias = new std::string("col1");
+    auto* ape = new AliasPropertyExpression(new std::string(""), alias, col);
+    std::string c0("col1_row_9999");
+    auto* pe = new PrimaryExpression(c0);
+    auto r =  std::make_unique<RelationalExpression>(ape,
+                                                     RelationalExpression::Operator::EQ,
+                                                     pe);
+    lookupExec(expectRowNum, Expression::encode(r.get()));
+}
+
+BENCHMARK(PreciseScan_10000) {
+    modeInt = 100;
+    lookupPrecise(10000);
+}
+
+BENCHMARK(PreciseScan_100) {
+    modeInt = 10000;
+    lookupPrecise(100);
+}
+
+BENCHMARK(PreciseScan_10) {
+    modeInt = 100000;
+    lookupPrecise(10);
+}
+
+BENCHMARK(FilterScan_1) {
+    modeInt = FLAGS_total_vertices_size;
+    lookupFilter(1);
+}
+
+}  // namespace storage
+}  // namespace nebula
+
+int main(int argc, char** argv) {
+    folly::init(&argc, &argv, true);
+    folly::runBenchmarks();
+    return 0;
+}
+
+
+/**
+ * CPU : Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
+ * MemTotal : 8G
+ * Total vertices : 1000000
+ * PreciseScan_10000 : Precise scan , return 10000 vertices
+ * PreciseScan_100   : Precise scan , return 100 vertices
+ * PreciseScan_10    : Precise scan , return 10 vertices
+ * FilterScan_10     : Full index scan, return 1 vertex
+ **/
+
+/**
```

Comment:
What's the index and expression?  
What's the part number?

Reply:
> What's the index and expression?
> What's the part number?

- Part number : 
1 part

- tag、index structure as below : 
```
tag (col0 string, col1 string, col2 string)
index on tag(col0, col1, col2)
```

- expression : 

```
PreciseScan : col0 == xxxx
FilterScan : col1 == xxxx
```
The FilterScan is not work now. I need to improve this benchmark at later.

Reply:
Fine.  Could you update the comments

Reply:
> Fine. Could you update the comments

Sure, comments updated.